apps/server/services/bus/handlers.go

```diff
@@ -444,28 +352,126 @@ func (b *Bus) handleKVRequest(ctx context.Context, req *protos.KVRequest) error
+	}
+
+	// If this is NOT a brand new audience, nothing to do
+	if len(existingPipelines) > 0 {
+		llog.Debugf("audience '%s' already has pipeline configuration - nothing to do", util.AudienceToStr(req.Audience))
+		return nil
+	}
+
+	// Get all session IDs on this node
+	sessionIDs, err := b.options.Store.GetSessionIDsByAudience(ctx, req.Audience, b.options.NodeName)
+	if err != nil {
+		llog.Errorf("error getting session ids by audience '%s' from store: %v", req.Audience, err)
+		return errors.Wrapf(err, "error getting session ids by audience '%s' from store", req.Audience)
+	}
+
+	if len(sessionIDs) == 0 {
+		llog.Debugf("no active sessions found for audience '%s' on node '%s' - skipping", req.Audience, b.options.NodeName)
+		return nil
+	}
+
+	// Send SetPipelines command to each session ID
+	if _, err := b.sendSetPipelinesCommand(req.Audience, make([]*protos.Pipeline, 0), sessionIDs...); err != nil {
+		llog.Errorf("unable to send SetPipelines command: %v", err)
+		return errors.Wrap(err, "error sending SetPipelines command")
+	}
+
+	// Inform the UI that changes have occurred
+	b.options.PubSub.Publish(types.PubSubChangesTopic, "changes detected via new audience broadcast handler")
+
+	llog.Debugf("sent SetPipelineCommands for '%d' sessions", len(sessionIDs))
+
 	return nil
 }
 
+// Wrapper for sending SetPipelines command to a list of session IDs; injects
+// schema inference pipeline into the list of provided pipelines. Returns
+// number of SetPipelines commands that were sent.
+func (b *Bus) sendSetPipelinesCommand(
+	aud *protos.Audience,
+	pipelines []*protos.Pipeline,
+	sessionIDs ...string,
+) (int, error) {
+	llog := b.log.WithFields(logrus.Fields{
+		"method": "sendSetPipelinesCommand",
+	})
+
+	// Inject schema inference pipeline
+	updatedPipelines, err := util.InjectSchemaInferenceForPipelines(pipelines, b.options.WASMDir)
+	if err != nil {
+		llog.Errorf("error injecting schema inference pipeline: %s", err)
+		return 0, errors.Wrap(err, "error injecting schema inference pipeline")
+	}
+
+	// Populate WASM
+	for _, p := range updatedPipelines {
+		// NOTE: Command_SetPipelines does not have a WasmModules field (like
+		// the internal.GetSetPipelinesCommandsByServiceResponse), so we need
+		// to fill out all of the wasm fields.
+		if err := util.PopulateWASMFields(p, b.options.WASMDir); err != nil {
+			llog.Errorf("error populating WASM fields: %s", err)
+			return 0, errors.Wrap(err, "error populating WASM fields")
+		}
+	}
+
+	var sent int
+
+	for _, sessionID := range sessionIDs {
+		ch := b.options.Cmd.GetChannel(sessionID)
+		if ch == nil {
+			llog.Errorf("expected cmd channel to exist for session id '%s' but none found - skipping", sessionID)
+			continue
+		}
+
+		llog.Debugf("sending SetPipelines command to session id '%s' on node '%s'", sessionID, b.options.NodeName)
+
+		// TODO: Need a more reliable writing mechanism - we can currently get
+		// stuck if ch goes away (SDK disconnects).
+
```

Comment:
Mark: this might just panic because the channel might be closed (due to client disconnect)